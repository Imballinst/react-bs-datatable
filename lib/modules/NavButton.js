"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var Button_1 = __importDefault(require("react-bootstrap/Button"));
function NavButton(_a) {
    var pageNumber = _a.pageNumber, disabled = _a.disabled, onPageNavigate = _a.onPageNavigate, label = _a.label, className = _a.className;
    var btnProps = {
        disabled: disabled,
        onClick: onPageNavigate(pageNumber),
        className: className
    };
    return react_1.default.createElement(Button_1.default, __assign({}, btnProps), label);
}
exports.default = NavButton;
//# sourceMappingURL=NavButton.js.map