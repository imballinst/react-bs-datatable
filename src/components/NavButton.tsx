import React from 'react';
import { TableComponents } from '../helpers/types';

type NavButtonProps = {
  pageNumber: number;
  disabled?: boolean;
  onPageNavigate: any;
  label: string | number;
  className?: string;
  Component: TableComponents['Button'];
};

export default function NavButton({
  pageNumber,
  disabled,
  onPageNavigate,
  label,
  className,
  Component,
  ...props
}: NavButtonProps) {
  const btnProps = {
    ...props,
    disabled,
    onClick: () => onPageNavigate(pageNumber),
    className
  };

  return <Component {...btnProps}>{label}</Component>;
}
