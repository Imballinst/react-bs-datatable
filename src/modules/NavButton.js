import React from 'react';
import PropTypes from 'prop-types';

import Button from 'react-bootstrap/Button';

function NavButton({ pageNavNumber, disabled, onPageNavigate, label }) {
  const btnProps = {
    disabled,
    onClick: onPageNavigate(pageNavNumber)
  };

  return <Button {...btnProps}>{label}</Button>;
}

NavButton.propTypes = {
  pageNavNumber: PropTypes.number.isRequired,
  disabled: PropTypes.bool.isRequired,
  onPageNavigate: PropTypes.func.isRequired,
  label: PropTypes.string.isRequired
};

export default NavButton;
