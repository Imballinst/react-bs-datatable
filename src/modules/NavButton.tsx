import React from 'react';

import Button from 'react-bootstrap/Button';

type NavButtonProps = {
  pageNumber: number;
  disabled: boolean;
  onPageNavigate: any;
  label: string;
};

export default function NavButton({
  pageNumber,
  disabled,
  onPageNavigate,
  label
}: NavButtonProps) {
  const btnProps = {
    disabled,
    onClick: onPageNavigate(pageNumber)
  };

  return <Button {...btnProps}>{label}</Button>;
}
