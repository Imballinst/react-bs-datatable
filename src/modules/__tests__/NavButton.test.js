import React from 'react';
import { shallow } from 'enzyme';

import NavButton from '../NavButton';

function setup() {
  const props = {
    pageNavNumber: 5,
    disabled: false,
    onPageNavigate: jest.fn(val => expect(val).toBe(5)),
    label: 'test-label'
  };

  const enzymeWrapper = shallow(<NavButton {...props} />);

  return {
    props,
    enzymeWrapper
  };
}

describe('NavButton component (src/modules/NavButton)', () => {
  it('should match normal snapshot', () => {
    const { props, enzymeWrapper } = setup();

    expect(enzymeWrapper).toMatchSnapshot();
  });

  it('should have the same props before and after render', () => {
    const { props, enzymeWrapper } = setup();

    expect(enzymeWrapper.prop('children')).toBe('test-label');
    expect(enzymeWrapper.prop('onClick')).toBe(undefined);
    expect(enzymeWrapper.prop('disabled')).toBe(false);
  });

  // FIXME: this test is broken.
  // it('should call the jest mock function onPageNavigate', () => {
  //   const { props, enzymeWrapper } = setup();

  //   expect(enzymeWrapper.prop('onPageNavigate').mock.calls.length).toBe(1);
  // });
});
