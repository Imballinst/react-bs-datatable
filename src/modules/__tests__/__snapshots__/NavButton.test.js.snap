// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavButton component (src/modules/NavButton) should match normal snapshot 1`] = `ShallowWrapper {}`;
